src/CppUTestExt/MockSupport_c.cpp

```diff
@@ -788,7 +788,7 @@ const void* returnConstPointerValueOrDefault_c(const void * defaultValue)
 
 void (*functionPointerReturnValue_c())()
 {
-    return actualCall->returnFunctionPointerValue();
+    return (void (*)()) actualCall->returnFunctionPointerValue();
```

Comment:
The cast has nothing to do with this PR - I checked the new code against the Cl2000 and it requires this cast. But it is also currently broken because of added features, and not easy to fix, so I can take this one out here, but there won't be a (working) PR for the Cl2000.